crates/driver/src/domain/mempools.rs

```diff
@@ -41,12 +41,13 @@ impl Mempools {
         &self,
         solver: &Solver,
         settlement: &Settlement,
+        submission_deadline: u64,
```

Comment:
nit: add `type BlockNo = u64;` to avoid confusion of what this u64 is here as well (could in theory also be a timestamp)